mlflow/pyfunc/model.py

```diff
@@ -391,19 +391,329 @@ def predict_stream(
+    **Logging the ChatAgent**
+
+    Since the landscape of LLM frameworks is constantly evolving and not every flavor can be natively supported by MLflow, we recommend a code-based pyfunc logging approach (docs link). Don’t forget to add resources if you are using any Databricks serving endpoints, tools, or vector search indexes!
+
+    .. code-block:: python
+
+        with mlflow.start_run():
+            logged_agent_info = mlflow.pyfunc.log_model(
+                artifact_path='agent',
+                python_model=os.path.join(os.getcwd(),"agent"),
+                pip_requirements=[
+                    "mlflow",
+                    "openai",
+                ],
+                resources=[<list resources here>],
+            )
+
+    After logging the model, you can query the model with a single dictionary, making this easy to serve. It will be converted into the 3 Pydantic models, which is what is expected by your ``predict`` and ``predict_stream`` methods.
+
+    .. code-block:: python
+
+        loaded_model = mlflow.pyfunc.load_model(tmp_path)
+        loaded_model.predict(
+            {
+                "messages": [{"role": "user", "content": "What is 10 + 10?"}],
+                "context": {"conversation_id": "123", "user_id": "456"},
+            }
+        )
+
+    To make logging ChatAgent models as easy as possible, MLflow has built in the following features:
+
+    - Model Signature
+        - You do not need to set a signature when logging a ChatAgent
+        -  An input and output signature will be automatically set that adheres to the :py:class:`ChatAgentRequest <mlflow.types.agent.ChatAgentRequest>` and :py:class:`ChatAgentResponse <mlflow.types.agent.ChatAgentResponse>` schemas
+    - Metadata
+        - ``{“task”: “agent/v2/chat”}`` will be automatically appended to any metadata that you may pass in when logging the model
+
+    - Input Example
+        - Providng an input example is optional, CHAT_AGENT_INPUT_EXAMPLE in mlflow.types.agent will be provided by default
+        - If you do provide an input example, ensure it's a dict with the :py:class:`ChatAgentRequest <mlflow.types.agent.ChatAgentRequest>` schema
+
+        - .. code-block:: python
+
+            input_example = {
+                "messages": [{"role": "user", "content": "What is MLflow?"}],
+                "context": {"conversation_id": "123", "user_id": "456"},
+            }
+
+    **Going from ChatModel to ChatAgent**
+
+    To convert an existing ChatModel that takes in :py:class:`ChatMessage <mlflow.types.llm.ChatMessage>` and :py:class:`ChatParams <mlflow.types.llm.ChatParams>` and outputs a :py:class:`ChatCompletionResponse <mlflow.types.llm.ChatCompletionResponse>`, do the following:
+
+    - Redefine your class
+    - Redefine functionality from load_context into each predict call because we removed the context param from ChatAgent's predict functionality.
+    - Call ``.model_dump_compat()`` instead of ``.to_dict()`` for the method args
```

Comment:
Maybe explain with a bit more context what's going on here  from the developer's perspective - something like "Use the `.model_dump_compat()` API instead of `.to_dict()` when converting the inputs to `predict` and `predict_stream` into dictionaries, e.g. `[msg.model_dump_compat() for msg in messages]`, rather than `[msg.to_dict() for msg in messages]`